src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/amendment/mappers/AmendmentToEdifactMapper.java

```diff
@@ -0,0 +1,49 @@
+package uk.nhs.digital.nhsconnect.nhais.translator.amendment.mappers;
+
+import org.apache.commons.lang3.StringUtils;
+import uk.nhs.digital.nhsconnect.nhais.exceptions.PatchValidationException;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.Segment;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentBody;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatch;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatchOperation;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.JsonPatches;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public abstract class AmendmentToEdifactMapper {
+
+    protected static boolean amendmentPatchRequiringValue(AmendmentPatch amendmentPatch) {
+        return amendmentPatch.getOp() == AmendmentPatchOperation.ADD
+            || amendmentPatch.getOp() == AmendmentPatchOperation.REPLACE;
+    }
+
+    public Optional<Segment> map(AmendmentBody amendmentBody) throws PatchValidationException {
+        var patches = amendmentBody.getJsonPatches();
+        validatePatches(patches);
+        if (shouldCreateSegment(amendmentBody)) {
+            return Optional.of(mapPatches(amendmentBody));
+        }
+        return Optional.empty();
+    }
+
+    abstract void validatePatches(JsonPatches patches) throws PatchValidationException;
+
+    abstract Segment mapPatches(AmendmentBody amendmentBody);
+
+    abstract boolean shouldCreateSegment(AmendmentBody amendmentBody);
+
+    protected void validateNonEmptyValues(List<Optional<AmendmentPatch>> amendmentPatches) {
+        var invalidAmendmentPaths = amendmentPatches.stream()
+            .flatMap(Optional::stream)
+            .filter(AmendmentToEdifactMapper::amendmentPatchRequiringValue)
+            .filter(amendmentPatch -> StringUtils.isBlank(amendmentPatch.getValue().get()))
+            .map(AmendmentPatch::getPath)
+            .collect(Collectors.toList());
+
+        if (!invalidAmendmentPaths.isEmpty()) {
+            throw new PatchValidationException("Invalid values for: " + invalidAmendmentPaths);
+        }
+    }
```

Comment:
This probably should be checked at `AmendmentController` level

Reply:
I prefer having all validation logic in one place. Let's leave it here for now

Reply:
We can leave it here, but `AmendmentController` is doing request validation also (so it might be in one place there ;) ). The second thing is that this logic isn't related to mapping specific data/segment - all mappers will do exactly the same work